dlt/common/configuration/exceptions.py

```diff
@@ -73,29 +73,34 @@ def __str__(self) -> str:
                     " missing or empty.\n"
                 )
 
-        # check if entry point is run with path. this is common problem so warn the user
         main_path = main_module_file_path()
-        if main_path and main_path.endswith(".py"):
+
+        # check if entry point is run with path or from cli command. This is common problem so warn the user
+        if main_path and (main_path.endswith(".py") or main_path.endswith("dlt")):
```

Comment:
We have two cases here to detect
1. pipeline context exists and we can get `last_run_context` from it. then we warn the user if settings_dir is different in pipeline than in current run context (it should be always available). the error message is the same as you wrote in the drop command.
2. else: the check that was already there. it was mostly warning if user wrote a pipeline, saved it into folder but then run it from a different cwd (for a first time)